src/dawn/engine_dawn.cc

```diff
@@ -1387,5 +1480,95 @@ Result EngineDawn::AttachBuffersAndTextures(
   return {};
 }
 
+Result EngineDawn::AttachBuffers(ComputePipelineInfo* compute_pipeline) {
+  Result result;
+
+  // Do not attach pushConstants
+  if (compute_pipeline->pipeline->GetPushConstantBuffer().buffer != nullptr) {
+    return Result("AttachBuffers: Dawn does not support push constants!");
+  }
+
+  std::vector<std::vector<BindingInitializationHelper>> bindingInitalizerHelper(
+      kMaxDawnBindGroup);
+  std::vector<std::vector<::dawn::BindGroupLayoutBinding>> layouts_info(
+      kMaxDawnBindGroup);
+  uint32_t max_descriptor_set = 0;
+
+  // Attach storage/uniform buffers
+  for (const auto& buf_info : compute_pipeline->pipeline->GetBuffers()) {
+    ::dawn::BufferUsageBit bufferUsage;
+    ::dawn::BindingType bindingType;
+    switch (buf_info.buffer->GetBufferType()) {
+      case BufferType::kStorage: {
+        bufferUsage = ::dawn::BufferUsageBit::Storage;
+        bindingType = ::dawn::BindingType::StorageBuffer;
+        break;
+      }
+      case BufferType::kUniform: {
+        bufferUsage = ::dawn::BufferUsageBit::Uniform;
+        bindingType = ::dawn::BindingType::UniformBuffer;
+        break;
+      }
+      default: {
+        return Result("AttachBuffers: unknown buffer type: " +
+                      std::to_string(static_cast<uint32_t>(
+                          buf_info.buffer->GetBufferType())));
+        break;
+      }
+    }
+
+    if (buf_info.descriptor_set > kMaxDawnBindGroup - 1) {
+      return Result(
+          "AttachBuffers: Dawn has maximum of 4 bindGroups "
```

Comment:
The 4 is magic here, and could go out of date.

Do this:

`std::string("AttachBuffers: Dawn has a maximum of ") + std::to_string(kMaxDawnBindGroup) + " bind groups `....